kube_apiserver_metrics/tests/test_sli_metrics.py

```diff
@@ -0,0 +1,130 @@
+        SLIMetric("slis.kubernetes_healthchecks_total", 5, ["sli_name:log", "status:success"]),
+        SLIMetric("slis.kubernetes_healthchecks_total", 5, ["sli_name:ping", "status:success"]),
+        SLIMetric(
+            "slis.kubernetes_healthchecks_total",
+            5,
+            ["sli_name:poststarthook/aggregator-reload-proxy-client-cert", "status:success"],
+        ),
+        SLIMetric(
+            "slis.kubernetes_healthchecks_total",
+            5,
+            ["sli_name:poststarthook/apiservice-discovery-controller", "status:success"],
+        ),
+    ]
+    for metric in expected_metrics:
+        aggregator.assert_metric("{}.{}".format(CHECK_NAME, metric.name), value=metric.value, tags=metric.tags)
+
+    aggregator.assert_all_metrics_covered()
+
+
+def test_check_metrics_slis_transform(aggregator, mock_metrics, instance):
+    c = KubeAPIServerMetricsCheck(CHECK_NAME, {}, [instance])
+    c.check(instance)
+
+    def assert_metric(name, **kwargs):
+        # Wrapper to keep assertions < 120 chars
+        aggregator.assert_metric("{}.{}".format(CHECK_NAME, name), **kwargs)
+
+    # Check that no metrics with `name` tag come through
+    assert_metric(
+        "slis.kubernetes_healthcheck",
+        count=0,
+        metric_type=aggregator.GAUGE,
+        tags=["name:autoregister-completion"],
+    )
+    assert_metric(
+        "slis.kubernetes_healthchecks_total",
+        metric_type=aggregator.MONOTONIC_COUNT,
+        count=0,
+        tags=["name:autoregister-completion", "status:error"],
+    )
+
+
+def test_check_metrics_slis_filter_by_type(aggregator, mock_metrics, instance):
+    c = KubeAPIServerMetricsCheck(CHECK_NAME, {}, [instance])
+    c.check(instance)
+
+    def assert_metric(name, **kwargs):
+        # Wrapper to keep assertions < 120 chars
+        aggregator.assert_metric("{}.{}".format(CHECK_NAME, name), **kwargs)
+
+    # Check that metrics with type other than `healthz` are filtered out
+    assert_metric(
+        "slis.kubernetes_healthcheck",
+        count=0,
+        metric_type=aggregator.GAUGE,
+        tags=["sli_name:autoregister-completion", "type:readyz"],
+    )
+
+    assert_metric(
+        "slis.kubernetes_healthchecks_total",
+        metric_type=aggregator.MONOTONIC_COUNT,
+        count=0,
+        tags=["sli_name:autoregister-completion", "status:error", "type:readyz"],
+    )
+
+
+def test_detect_sli_endpoint(mock_metrics, instance):
+    c = KubeAPIServerMetricsCheck(CHECK_NAME, {}, [instance])
+    assert c._slis_available is True
+
+
+def test_detect_sli_endpoint_404(instance, mock_http_response):
+    mock_http_response(status_code=404)
+    c = KubeAPIServerMetricsCheck(CHECK_NAME, {}, [instance])
+    assert c._slis_available is False
+
+
+def test_detect_sli_endpoint_403(instance, mock_http_response):
+    mock_http_response(status_code=403)
+    c = KubeAPIServerMetricsCheck(CHECK_NAME, {}, [instance])
+    assert c._slis_available is False
```

Comment:
Is there a way to test **behavior** here instead of the implementation? I mean, will `_slis_available` have some effect on the metrics collected?

Reply:
changed this to test for the count of metrics instead (which should be 0 if the endpoint is unavailable)!

Reply:
nice!